tests/system/src/test/java/test/robot/javafx/scene/ChoiceBoxScrollUpOnCollectionChangeTest.java

```diff
@@ -95,11 +94,11 @@ private void scrollChoiceBox(int scrollAmt) throws Exception {
         Util.runAndWait(() -> {
             for (int i = 0; i < scrollAmt; i++) {
                 robot.keyType(KeyCode.DOWN);
-                Toolkit.getToolkit().firePulse();
             }
             scrollLatch.countDown();
         });
 
+        Util.waitForIdle(scene);
```

Comment:
is this the right place for waitForIdle?  should it be on L98? since there are multiple key presses involved?

and also, do we still need to .sleep() on L103?

Reply:
Ideally yes it should be on L98.  Since we can not call `waitForIdle` from Application thread I added at L101. But if we have to achieve the purpose of using `waitForIdle` in the test then it should be added after each key press. So updated the code to do so.
I had kept the sleep as it is thinking if it might be required in slower systems. Since we are waiting  for 10 pulses with `waitForIdle` it might not be needed. Hence removed it.
Since we are using `Util.runAndWait` there is no need for the `scrollLatch` hence removed it.